hasher-matcher-actioner/hmalib/models.py

```diff
@@ -97,12 +98,44 @@ def to_dynamodb_item(self) -> dict:
             "PK": self.get_dynamodb_signal_key(self.signal_source, self.signal_id),
             "SK": self.get_dynamodb_ds_key(self.ds_id),
             "SignalHash": self.signal_hash,
-            "SignalSource": self.signal_source,  # defaults to 'te' in the current pipeline
+            "SignalSource": self.signal_source,
             "UpdatedAt": self.updated_at.isoformat(),
             "HashType": self.SIGNAL_TYPE,
             "Tags": self.tags,
         }
 
+    def update_tags_in_table_if_exists(self, table: Table) -> bool:
```

Comment:
I'm going to flag that this seems like a good opportunity to use HMAConfig (though admittedly not a config, maybe we could change the name to HMAS3Adapter)

We could get rid of most of this boiler plate. It would change this fucntion to:
```
signal = PDQSignalMetadata.get(self.signal_source, self.signal_id)
signal.tags = new_tags
PDQSignalMetadata.update(signal)
```

Reply:
For objects that use the single table model of the HMA-Datastore table I am more opposed to having it be dynamically generated like HMAConfig, indexes, typing, etc. become more complex so it is hard for me to see right now if tradeoff would be worth it.

Reply:
Indexes seems like a valid reason not to pursue right now. just planting the seed